datalad/support/annexrepo.py

```diff
@@ -297,6 +297,21 @@ def add_remote(self, name, url, options=None):
                              " -S %s" % c.ctrl_path)
             writer.release()
 
+    def set_remote_url(self, name, url, push=False):
+        """Overrides method from GitRepo in order to set
+        remote.<name>.annex-ssh-options in case of a SSH remote."""
+
+        super(AnnexRepo, self).set_remote_url(name, url, push=push)
+        from datalad.support.network import is_ssh
+        if is_ssh(url):
+            c = ssh_manager.get_connection(url)
```

Comment:
theoretically such constructs should be within try/finally
```python
try:
  get connection
finally:
  release
```
otherwise if something booms in writer but caught upstairs and "handled" we might be having a dangling thing with unknown consequences